backend/src/main/java/com/javahelp/backend/TokenAuthManager.java

```diff
@@ -0,0 +1,37 @@
+package com.javahelp.backend;
+
+import com.javahelp.backend.data.ITokenStore;
+import com.javahelp.backend.data.IUserStore;
+import com.javahelp.model.token.Token;
+import com.javahelp.model.user.User;
+
+public class TokenAuthManager {
+    private final IUserStore userStore;
+    private final ITokenStore tokenStore;
+    private TokenAuthResult authenticate;
+
+    /**
+     * Constructs a TokenAuthManager object.
+     *
+     * @param userStore: interface that manages users
+     * @param tokenStore: interface that manages tokens
+     */
+    public TokenAuthManager(IUserStore userStore, ITokenStore tokenStore) {
+        this.userStore = userStore;
+        this.tokenStore = tokenStore;
+    }
+
+    public void setAuthenticate(User desiredUser, Token token) {
+        this.authenticate = new TokenAuthResult(desiredUser, token);
+    }
+
+    /**
+     * compares the input user and token and returns the result of the comparison
+     *
+     * @return TokenAuthResult object with the results of the comparison
+     */
+    public TokenAuthResult authenticate(User desiredUser, Token token) {
```

Comment:
Can this method be combined with `setAuthenticate()`? Since it's just one line of code in `setAuthenticate`. If you do combine, would it be better if we can just return the instance of `TokenAuthResult` directly instead of setting it as an attribute of the class and then returning it?

Reply:
Yeah I can definitely combine the two methods, I think we could just return an instance of TokenAuthResult but the current version is how it's done in the CRC cards so I'm not sure if it'll come into play later

Reply:
I think it would probably be better to keep the `TokenAuthResult` as a local field, and not as a field in the class.